public/app/features/dashboard/timepicker/timepicker.ts

```diff
@@ -96,13 +96,12 @@ export class TimePickerCtrl {
   }
 
   openDropdown() {
+    this.$rootScope.appEvent('escTimepicker');
```

Comment:
This looks weird. The timepicker should not care about if esc or any other key is pressed. We want the timepicker to be as dumb as possible and only care about it's internal functionality.